### ISSUE TITLE
conda deactivate generates traceback in some cases

### ISSUE THREAD

**Issue by wshanks** [xonsh/xonsh#4058]
I think the description could be "conda deactivate generates traceback for some deactivate scripts" but I am not certain that is the problem.

## xonfig

<details>

```
+------------------+-----------------+
| xonsh            | 0.9.24          |
| Python           | 3.8.5           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 3.0.10          |
| shell type       | prompt_toolkit  |
| pygments         | 2.7.3           |
| on posix         | True            |
| on linux         | True            |
| distro           | unknown         |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
| on jupyter       | False           |
| jupyter kernel   | None            |
+------------------+-----------------+
```

</details>

## conda env export -n base

<details>

```
name: base                                                                                                                                                                                   
channels:                                                                                                                                                                                    
  - conda-forge                                                                                                                                                                              
  - https://conda.anaconda.org/conda-forge                                                                                                                                                   
dependencies:                                                                                                                                                                                
  - _libgcc_mutex=0.1=conda_forge                                                                                                                                                            
  - _openmp_mutex=4.5=1_gnu                                                                                                                                                                  
  - backports=1.0=py_2         
  - backports.functools_lru_cache=1.6.1=py_0
  - beautifulsoup4=4.9.3=pyhb0f4dca_0
  - brotlipy=0.7.0=py38h497a2fe_1001
  - bzip2=1.0.8=h516909a_2                
  - c-ares=1.17.1=h36c2ea0_0 
  - ca-certificates=2020.12.5=ha878542_0
  - certifi=2020.12.5=py38h578d9bd_1
  - cffi=1.14.4=py38ha312104_0
  - chardet=4.0.0=py38h578d9bd_1  
  - conda=4.9.2=py38h578d9bd_0  
  - conda-build=3.20.3=py38h32f6830_0
  - conda-package-handling=1.7.2=py38h8df0ef7_0 
  - conda-suggest=0.1.1=pyh9f0ad1d_0     
  - conda-suggest-conda-forge=2020.11.25=ha770c72_0
  - cryptography=3.3.1=py38h2b97feb_1
  - filelock=3.0.12=pyh9f0ad1d_0
  - glob2=0.7=py_0          
  - icu=68.1=h58526e2_0     
  - idna=2.10=pyh9f0ad1d_0    
  - jinja2=2.11.2=pyh9f0ad1d_0        
  - krb5=1.17.2=h926e7f8_0          
  - ld_impl_linux-64=2.35.1=hea4e1c9_1
  - libarchive=3.5.1=h3f442fb_1  
  - libcurl=7.71.1=hcdd3856_8      
  - libedit=3.1.20191231=he28a2e2_2
  - libev=4.33=h516909a_1      
  - libffi=3.2.1=he1b5a44_1007   
  - libgcc-ng=9.3.0=h5dbcf3e_17 
  - libgomp=9.3.0=h5dbcf3e_17      
  - libiconv=1.16=h516909a_0              
  - liblief=0.10.1=he1b5a44_2
  - libnghttp2=1.41.0=h8cfc5f6_2
  - libsolv=0.7.16=h8b12597_0
  - libssh2=1.9.0=hab1572f_5
  - libstdcxx-ng=9.3.0=h2ae2ef3_17
  - libxml2=2.9.10=h72842e0_3
  - lz4-c=1.9.3=h9c3ff4c_0      
  - lzo=2.10=h516909a_1000   
  - mamba=0.7.8=py38h2aa5da1_1           
  - markupsafe=1.1.1=py38h497a2fe_3      
  - ncurses=6.2=h58526e2_4          
  - openssl=1.1.1i=h7f98852_0
  - patchelf=0.11=he1b5a44_0
  - pip=20.2=py_0           
  - pkginfo=1.6.1=pyh9f0ad1d_0
  - prompt-toolkit=3.0.10=pyha770c72_0
  - prompt_toolkit=3.0.10=hd8ed1ab_0
  - psutil=5.8.0=py38h497a2fe_1
  - py-lief=0.10.1=py38h348cfbe_2
  - pycosat=0.6.3=py38h497a2fe_1006
  - pycparser=2.20=pyh9f0ad1d_2
  - pygments=2.7.3=pyhd8ed1ab_0
  - pyopenssl=20.0.1=pyhd8ed1ab_0
  - pysocks=1.7.1=py38h578d9bd_3
  - python=3.8.5=h1103e12_5_cpython
  - python-libarchive-c=2.9=py38h578d9bd_2
  - python_abi=3.8=1_cp38  
  - pytz=2020.5=pyhd8ed1ab_0
  - pyyaml=5.3.1=py38h497a2fe_2
  - readline=8.0=he28a2e2_2
  - reproc=14.2.1=h36c2ea0_0
  - reproc-cpp=14.2.1=h58526e2_0
  - requests=2.25.1=pyhd3deb0d_0
  - ripgrep=12.1.1=h516909a_1
  - ruamel_yaml=0.15.80=py38h497a2fe_1003
  - setproctitle=1.1.10=py38h497a2fe_1004
  - setuptools=49.6.0=py38h578d9bd_3
  - six=1.15.0=pyh9f0ad1d_0
  - soupsieve=2.0.1=py_1
  - sqlite=3.34.0=h74cdb3f_0
  - tk=8.6.10=h21135ba_1
  - tqdm=4.56.0=pyhd8ed1ab_0
  - urllib3=1.26.2=pyhd8ed1ab_0
  - wcwidth=0.2.5=pyh9f0ad1d_2
  - wheel=0.36.2=pyhd3deb0d_0
  - xonsh=0.9.24=py38h578d9bd_0
  - xz=5.2.5=h516909a_1
  - yaml=0.2.5=h516909a_0
  - zlib=1.2.11=h516909a_1010
  - zstd=1.4.8=ha95c52a_1
  - pip:
    - xontrib-fzf-widgets==0.0.3
 ```
</details>

## Expected Behavior
`conda activate` and `conda deactivate` should work without printing tracebacks.

## Current Behavior
When I activate and then deactivate a conda environment with geotiff, I get a traceback about PWD not being defined in the current environment. The traceback does not crash the shell.

### Traceback (if applicable)

<details>

```
prompt: error: on field 'cwd'
Traceback (most recent call last):
  File "/home/wshanks/conda/lib/python3.8/site-packages/xonsh/prompt/base.py", line 90, in _no_cache_field_value
    value = field_value() if callable(field_value) else field_value
  File "/home/wshanks/conda/lib/python3.8/site-packages/xonsh/prompt/cwd.py", line 52, in _dynamically_collapsed_pwd
    original_path = _replace_home_cwd()
  File "/home/wshanks/conda/lib/python3.8/site-packages/xonsh/prompt/cwd.py", line 32, in _replace_home_cwd
    return _replace_home(builtins.__xonsh__.env["PWD"])
  File "/home/wshanks/conda/lib/python3.8/site-packages/xonsh/environ.py", line 1985, in __getitem__
    raise KeyError(e.format(key))
KeyError: 'Unknown environment variable: $PWD'
```

</details>

## Steps to Reproduce

1. `conda create -n test geotiff`
2. `conda activate test`
3. `conda deactivate`

Note: I first tried with `proj` (a dependency of `geotiff`) instead of `geotiff` but didn't see problems with just that. With `geotiff`, there are two activate/deactivate scripts, one for each of `geotiff` and `proj`.

Note: I see the traceback even with `$PROMPT` set to `'{BOLD_GREEN}{user}@{hostname}{BOLD_BLUE} {RESET} {BOLD_BLUE}{prompt_end}{RESET} '`. Usually, I have `{cwd_base}` in the prompt as well, but the issue is present whether or not it is there.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by gforsyth** [comment]
Hey @wshanks -- thanks for reporting.  This is weird one!

I am not able to reproduce the error using your reproducer -- the environment with `geotiff` activates and deactivates without issue.

Can you post your `xonshrc` file?  I'm curious if there's some weirdness in the `conda initialize` block

**Comment by wshanks** [comment]
Thanks, @gforsyth! I hadn't thought to test in a clean xonshrc file. I was able to find the problem by debugging my `xonshrc` file.

In case you're curious, here is what was happening. In my `xonshrc` file, I modified `source-bash` to work around some of the things I mentioned in #3734 like this:

```
def _source_bash(args, stdin, stdout, stderr):
    shlvl = $SHLVL
    aliases["source-foreign"](
        [
            "bash",
            "--sourcer=source",
            "--extra-args=--norc",
        ] + args,
        stdin,
        stdout,
        stderr,
    )
    $SHLVL = shlvl
aliases["source-bash"] = _source_bash
```

The `--norc` part was the issue as that led to `PWD` not being defined in the bash shell that does the sourcing. I changed that line to use `--rcfile=bashrc_xonsh` instead, where `bashrc_xonsh` is a vanilla bashrc file that just sources the OS default files. I had switched to `--norc` to avoid pulling in other settings from my main bashrc that conflict with my xonshrc.
